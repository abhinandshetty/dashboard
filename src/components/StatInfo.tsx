import React, { FC, ReactNode } from 'react'
import styled from '@emotion/styled'

const Box = styled.div`
  -webkit-box-shadow: 4px 8px 29px -4px rgba(0, 0, 0, 0.2);
  -moz-box-shadow: 4px 8px 29px -4px rgba(0, 0, 0, 0.2);
  box-shadow: 4px 8px 29px -4px rgba(0, 0, 0, 0.2);
  padding: 10px 20px;
`

const Label = styled.div`
  margin: 10px 0;
`

const Value = styled.div`
  margin: 10px 0;
  font-size: 28px;
  font-weight: 600;
`

const StatInfo: FC<{ label: ReactNode; value: string | number }> = ({ label, value }) => {
  return (
    <Box>
      <Label data-testid="label">{label}</Label>
      <Value data-testid="value">{value}</Value>
    </Box>
  )
}

export default StatInfo
