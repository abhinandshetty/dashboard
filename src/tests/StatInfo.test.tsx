import { render, waitFor, screen } from '@testing-library/react'
import StatInfo from 'components/StatInfo'
import renderer from 'react-test-renderer'

it('StatInfo component renders correctly', () => {
  const tree = renderer.create(<StatInfo label="Total clicks" value="12" />).toJSON()
  expect(tree).toMatchSnapshot()
})

test('StatInfo component renders data as expected', async () => {
  render(<StatInfo label="Total clicks" value="12" />)

  const label = await waitFor(() => screen.findByTestId('label'))
  expect(label).toBeInTheDocument()
  expect(label.innerHTML).toEqual('Total clicks')

  const value = await waitFor(() => screen.findByTestId('value'))
  expect(value).toBeInTheDocument()
  expect(value.innerHTML).toEqual('12')
})
