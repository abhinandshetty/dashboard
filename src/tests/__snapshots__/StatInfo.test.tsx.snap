// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatInfo component renders correctly 1`] = `
<div
  className="css-1xpc2y8"
>
  <div
    className="css-5od3bw"
    data-testid="label"
  >
    Total clicks
  </div>
  <div
    className="css-1dmgjwf"
    data-testid="value"
  >
    12
  </div>
</div>
`;
